docs/source/usage_guides/mps.mdx

```diff
@@ -0,0 +1,100 @@
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Accelerated PyTorch Training on Mac
+
+With PyTorch v1.12 release, developers and researchers can take advantage of Apple silicon GPUs for significantly faster model training. 
+This unlocks the ability to perform machine learning workflows like prototyping and fine-tuning locally, right on Mac.
+Apple's Metal Performance Shaders (MPS) as a backend for PyTorch enables this and can be used via the new `"mps"` device. 
+This will map computational graphs and primitives on the MPS Graph framework and tuned kernels provided by MPS.
+For more information please refer official documents [Introducing Accelerated PyTorch Training on Mac](https://pytorch.org/blog/introducing-accelerated-pytorch-training-on-mac/)
+and [MPS BACKEND](https://pytorch.org/docs/stable/notes/mps.html).
+
+### Benefits of Training and Inference using Apple M1 Chips
+
+1. Enables users to train larger networks or batch sizes locally
+2. Reduces data retrieval latency and provides the GPU with direct access to the full memory store due to unified memory architecture. 
+Therefore, improving end-to-end performance.
+3. Reduces costs associated with cloud-based development or the need for additional local GPUs.
+
+### Pre-requisites: Installing torch with mps support
+
+Following this great medium article [GPU-Acceleration Comes to PyTorch on M1 Macs](https://medium.com/towards-data-science/gpu-acceleration-comes-to-pytorch-on-m1-macs-195c399efcc1),
+below are the steps to make sure torch is installed with `mps` support.
+
+```
+# installing torch with m1 support on mac
+# install latest python from https://www.python.org/downloads/release/python-3106/
+# check the platform
+import platform
+platform.platform()
+'macOS-12.5-arm64-arm-64bit' 
+# This is compatible as the macOS version is above 12.3 and it is the ARM64 native version of Python
+# install torch 1.12 via the below command
+# pip3 install torch torchvision torchaudio
+# test the `mps` device support
+import torch
+torch.has_mps
+True
+a = torch.Tensor([10,11])
+a.to("mps")
+/Users/mac/ml/lib/python3.10/site-packages/torch/_tensor_str.py:103: UserWarning: The operator 'aten::bitwise_and.Tensor_out' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at  /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:11.)
+  nonzero_finite_vals = torch.masked_select(tensor_view, torch.isfinite(tensor_view) & tensor_view.ne(0))
+tensor([10.0000, 11.0000], device='mps:0')
+```
```

Comment:
We can break this up a bit to make it more digestible/readable + syntax highlighting

```suggestion
```python
# installing torch with m1 support on mac
# install latest python from https://www.python.org/downloads/release/python-3106/
# check the platform
import platform
platform.platform()
```
```python out
'macOS-12.5-arm64-arm-64bit' 
```
```python
# This is compatible as the macOS version is above 12.3 and it is the ARM64 native version of Python
# install torch 1.12 via the below command
# pip3 install torch torchvision torchaudio
# test the `mps` device support
import torch
torch.has_mps
```
```python out
True
```
```python
a = torch.Tensor([10,11])
a.to("mps")
```
```python out
/Users/mac/ml/lib/python3.10/site-packages/torch/_tensor_str.py:103: UserWarning: The operator 'aten::bitwise_and.Tensor_out' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at  /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:11.)
  nonzero_finite_vals = torch.masked_select(tensor_view, torch.isfinite(tensor_view) & tensor_view.ne(0))
tensor([10.0000, 11.0000], device='mps:0')
```
```

Reply:
I feel the above block is fine as it might be too distracting to have multiple blocks which can break the flow when the user is reading

Reply:
If you're using comments inside a code-block in the documentation, you are doing something wrong; you can put the comments as sentences **then** enter a code-block. Also you should use the doctest syntax to highlight the inputs and outputs, as it's not clear otherwise.

Reply:
Oh, okay, then maybe I can just leave it at the link to the medium blogpost instead of summarizing it this way.

Reply:
Done. Removed this entire block and just left it with a link to the medium blogpost explaining this in detail.